ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,86 @@ def test_add_host_with_null_password(self, host):
         host.track_create()
         host.check_create(result)
 
+    @staticmethod
+    def modify_config_maxhostname(host_tracker, value):
+        try:
+            command = host_tracker.make_command(
+                'config_mod',
+                **dict(
+                    setattr=u'ipamaxhostnamelength={}'.format(value)))
+            command()
+        except errors.EmptyModlist:
+            pass
+
+    @staticmethod
+    def generate_hostname(total_length):
+        if total_length < 9:
+            raise ArithmeticError("Total DNS length in theses tests"
+                                  "must be at least 9")
+        return "aaa.bbb.{}".format((total_length - 8) * "c")
```

Comment:
Sorry I missed this point earlier, but here you may create a label > 63 chars and hence you won't be testing the same thing. You should create a hostname with the provided length but with each label < 63 chars => add more labels instead of adding more chars to the top-level labal.